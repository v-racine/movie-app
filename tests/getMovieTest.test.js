const { Config } = require('../src/config/config');

Config.Get(process.env);

const AppFactory = require('../src/app');
const request = require('supertest');

describe('get a movie', () => {
  const mockPgClient = {};

  const app = AppFactory({
    pgClient: mockPgClient,
  });

  describe('when: users want to view ONE movie', () => {
    let rsp;

    let id = 2;

    let movies = {
      rows: [
        {
          id: 2,
          movie_title: 'Notorious',
          movie_year: 1946,
          run_time: 101,
        },
      ],
    };

    beforeEach(async () => {
      mockPgClient.dbQuery = jest.fn().mockImplementation(() => {
        return new Promise((resolve) => {
          resolve(movies);
        });
      });

      rsp = await request(app).get(`/movies/${id}`).send();
    });

    afterEach(() => {
      jest.clearAllMocks();
    });

    test('then: we return all the information about that ONE movie', async () => {
      expect(mockPgClient.dbQuery).toHaveBeenCalledWith(
        `SELECT * FROM movies WHERE id = $1`,
        id.toString(),
      );

      const text = rsp.text;
      expect(text).toMatchSnapshot();

      const status = rsp.status;
      expect(status).toBe(200);
    });
  });
});
