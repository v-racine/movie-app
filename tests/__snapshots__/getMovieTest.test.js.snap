// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`get a movie when: users want to view ONE movie then: we return all the information about that ONE movie 1`] = `
<!DOCTYPE html>
<html lang="en-US">
  <head>
    <title>
      The Ultimate Movie App
    </title>
    <meta charset="UTF-8">
    <link rel="stylesheet"
          href="/stylesheets/application.css"
    >
  </head>
  <body>
    <header>
      <h1>
        The Ultimate Movie App
      </h1>
    </header>
    <main>
      <p>
        Here is your selected movie:
      </p>
      <table>
        <tbody>
          <tr>
            <td>
              <b>
                Title:
              </b>
              Notorious
            </td>
          </tr>
          <tr>
            <td>
              <b>
                Year:
              </b>
              1946
            </td>
          </tr>
          <tr>
            <td>
              <b>
                Run-Time:
              </b>
              101 minutes
            </td>
          </tr>
        </tbody>
      </table>
      <footer>
        <form action="/movies/update/2"
              method="get"
        >
          <button>
            Edit Movie
          </button>
        </form>
        <form action="/movies/delete/2"
              method="post"
        >
          <button type="submit">
            Delete Movie
          </button>
        </form>
      </footer>
    </main>
  </body>
</html>
`;
